internal/storage/v2/clickhouse/tracestore/dbmodel/from_dbmodel.go

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package dbmodel
+
+import (
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/internal/jptrace"
+)
+
+func FromDBModel(dbTrace Trace) ptrace.Traces {
+	trace := ptrace.NewTraces()
+	resourceSpans := trace.ResourceSpans().AppendEmpty()
+	scopeSpans := resourceSpans.ScopeSpans().AppendEmpty()
+	span := scopeSpans.Spans().AppendEmpty()
+
+	resourceAttributes, err := AttributesGroupToMap(dbTrace.Resource.Attributes)
+	if err != nil {
+		jptrace.AddWarnings(span, fmt.Sprintf("failed to decode bytes value: %v", err))
```

Comment:
nit: there's no indication at this point in the code that the error is about "failed to decode bytes". Such error can be returned near the code that actually attempted decoding byte values. The most we can say here is "failed to decode attributes: %w"